### PR TITLE
Add some Build Instructions for some Clusters

### PR DESCRIPTION
**Pull request by davschneller**
We add instructions for:

* Leonardo
* LUMI (to be tested/optimized)
* SuperMUC NG Phase 2 (still highly experimental)

All tentative and subject to testing.

### CODE REVIEW

**Review by Thomas-Ulrich — APPROVED** [review]
Nice!
What are the current bottlenecks preventing a spack workflow?

**Comment by davschneller** [comment]
Let's say... A bit more experimenting could already do it. :D 
I'm still waiting a bit for https://github.com/SeisSol/SeisSol/tree/davschneller/tensorforge to become ready though—then we wouldn't need chainforge/gemmforge anymore.

File: Documentation/leonardo.rst [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+.. code-block:: bash
+
+    export GCC_ROOT=$(dirname $(dirname $(which gcc)))
+    export CUDA_ROOT=$(dirname $(dirname $(which nvcc)))
+
+    wget https://github.com/llvm/llvm-project/archive/refs/tags/llvmorg-16.0.6.tar.gz
+    tar -xf llvmorg-16.0.6.tar.gz
+    mkdir -p llvm-project-llvmorg-16.0.6/build
+    cd llvm-project-llvmorg-16.0.6/build
+    cmake ../llvm -GNinja -DCMAKE_INSTALL_PREFIX=$SEISSOL_PREFIX -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra;compiler-rt;openmp;polly" -DGCC_INSTALL_PREFIX="${GCC_ROOT}" -DCUDA_TOOLKIT_ROOT_DIR="${CUDA_ROOT}" -DLLVM_TARGETS_TO_BUILD="X86;NVPTX"
```
Comment (ravil-mobile) [inline Documentation/leonardo.rst:54]: You can also use to `lld` linker for building `llvm`. It is usually faster. Just add the following cmake option `-DLLD_BUILD_TOOLS=ON`. But, yes, you must be sure that `lld` is installed at the first place.

Comment (davschneller) [reply Documentation/leonardo.rst:54]: Hi, thanks! I'll add it in the next update for the instructions (maybe we can also avoid re-building LLVM entirely somehow...).

**Pull request merged**